Ionic updated

cmreddy071 opened the pull request:
Created ionic sortable-list example
- Written test cases
-

joebochill commented on src/app/app.component.html:
```diff
@@ -1,3 +1,39 @@
+    <ion-toolbar>
+      <ion-title style="text-align: center; padding: 0; font-size: 16px;">
+        Sortable list
+        <mat-icon style="float: right; margin-right: 15px;" id="edit-list" (click)="edit()">
```
refer to loading-states for the styles to use for the toolbar layout

joebochill commented on src/app/app.component.scss:
```diff
@@ -0,0 +1,21 @@
+
```
can we find a way to eliminate these !important rules?

joebochill commented on src/app/app.component.spec.ts:
```diff
@@ -42,6 +42,18 @@ describe('AppComponent', () => {
+    app.draggable = true;
+    expect(app.draggable).toBe(true);
+  });
 
```
can we add a test that checks that the re-ordering works properly?

joebochill commented on src/app/app.component.ts:
```diff
@@ -24,4 +46,18 @@ export class AppComponent {
+  }
+  edit() {
+    this.draggable = !this.draggable;
+    console.log(this.draggable);
```
remove console.log statements

joebochill commented on src/app/app.module.ts:
```diff
@@ -1,22 +1,39 @@
 
+import { NgModule,  } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { FormsModule } from '@angular/forms';
```
are we using this?

joebochill commented on src/global.scss:
```diff
@@ -13,14 +13,43 @@
+  --background-hover : map-get($pxb-blue, 500);
+}
+
+.mat-list-base .mat-list-item.mat-list-item-with-avatar, .mat-list-base .mat-list-option.mat-list-item-with-avatar {
```
sketchy style rule

cmreddy071 commented:
Requested Changes are done.
- Changed drag and drop look.
- Added a test that checks re-ordering completed.
- Removed unused imports.
- Removed console.log Statements.

joebochill commented on src/app/app.component.html:
```diff
@@ -1,16 +1,16 @@
   <ion-header>
-    <ion-toolbar>
-      <ion-title style="text-align: center; padding: 0; font-size: 16px;">
+    <ion-toolbar mode="ios">
```
why mode ios? Why not the flex layout we use in the other examples?

cmreddy071 replied:
changed the mode ios.

joebochill commented on src/app/app.component.scss:
```diff
@@ -1,11 +1,10 @@
-
+  display: none;
 
   mat-list-item{
```
no longer need this rule at all if we are doing display:none on the preview

cmreddy071 replied:
removed.

joebochill commented on src/app/app.component.scss:
```diff
@@ -19,3 +18,20 @@
+}
+.cdk-drag-placeholder {
+  background: #ffffff;
+  box-shadow: 0 4px 16px rgba(0,0,0,.12);
```
where did these values come from?

cmreddy071 replied:
removed shadow values.

joebochill commented on src/app/app.component.scss:
```diff
@@ -19,3 +18,20 @@
+}
+.cdk-drag-animating {
+  transition: transform 550ms cubic-bezier(0, 0, 0.2, 1);
+}
```
Can we drop these to 250ms like the Angular Material example: https://material.angular.io/cdk/drag-drop/overview#reordering-lists

cmreddy071 replied:
dropped to 250ms.

joebochill commented on src/app/app.component.scss:
```diff
@@ -19,3 +18,20 @@
+
+.mat-list-base  {
+  .mat-list-item.mat-list-item-with-avatar{
+    height: 48px;
```
48 - not divisible by 5

joebochill commented on src/app/app.component.spec.ts:
```diff
@@ -42,6 +42,25 @@ describe('AppComponent', () => {
+    app.draggable = true;
+    expect(app.draggable).toBe(true);
+  });
 
```
This is an exact duplicate of the test above and does not validate that the reordering is successful.

cmreddy071 replied:
written test case for reorders the list successfully.

joebochill commented on src/app/app.component.scss:
```diff
@@ -2,36 +2,24 @@
 }
 .cdk-drag-placeholder {
-  background: #ffffff;
-  box-shadow: 0 4px 16px rgba(0,0,0,.12);
```
We still want to have a shadow, I was just curious where you got the values from?

joebochill commented on src/app/app.component.html:
```diff
@@ -1,3 +1,39 @@
+          </div>
+          <div>
+            {{ i.details }}
+          </div>
```
We should use Angular Typography elements for this instead of styling a div manually

joebochill commented on src/app/app.component.html:
```diff
@@ -1,3 +1,39 @@
+          </div>
+          <div>
+            {{ i.details }}
+          </div>
```
Use angular typography

cmreddy071 replied:
used angular typograpgy

cmreddy071 replied:
used angular typography.

joebochill commented on src/app/app.component.scss:
```diff
@@ -0,0 +1,25 @@
+  transition: transform 250ms cubic-bezier(0, 0, 0.2, 1);
+}
+.cdk-drag-placeholder {
+  background: #ffffff;  
```
still want a shadow on this...I was just wondering where you got the values for it from.

joebochill commented on src/app/app.component.scss:
```diff
@@ -0,0 +1,25 @@
+  }
+
+.cdk-drop-dragging .cdk-drag {
+  transition: transform 250ms cubic-bezier(0, 0, 0.2, 1);
```
Still wondering about this delay. Angular's demo has a noticable delay (onDrop) as well, but I wonder if dropping this value down to 100ms or 50ms will have any effect on that.

cmreddy071 replied:
onDrop delay is fixed.

joebochill replied:
Change these animation durations from (top to bottom): 250ms, 250ms, 50ms. That will make the drop snappier, but keep the smoothness on the drag.

joebochill commented:
Weird black box when dropping items:
![Simulator Screen Shot - iPhone Xʀ - 2019-10-17 at 12 06 06](https://user-images.githubusercontent.com/29152776/67027416-83b7b380-f0d7-11e9-904e-5f267cb1a489.png)

joebochill commented on src/app/app.component.spec.ts:
```diff
@@ -42,6 +42,27 @@ describe('AppComponent', () => {
+  it('Enable dragging', async () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.debugElement.componentInstance;
+    app.draggable = true;
```
instead of setting draggable to true and then testing is draggable is true, can you use the click event for the button?

cmreddy071 commented:
-Changed the animation durations from (top to bottom): 250ms, 250ms, 50ms.
- Test cases modified.
